docs/sidebars.js

```diff
@@ -1,152 +1,80 @@
-/**
- * Creating a sidebar enables you to:
- - create an ordered group of docs
- - render a sidebar for each doc of that group
- - provide next/previous navigation
-
- The sidebars can be generated from the filesystem, or explicitly defined here.
-
- Create as many sidebars as you want.
- */
-
 // @ts-check
+import fs from 'node:fs';
 
-/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
-const sidebars = {
-  // By default, Docusaurus generates a sidebar from the docs folder structure
-  // tutorialSidebar: [{type: 'autogenerated', dirName: '.'}],
-
-  // But you can create a sidebar manually
-  sidebarNode: [
+async function sleep(timeout) {
+  return new Promise((res) => {
+    setTimeout(res, timeout);
+  });
+}
+async function waitAndImportFile(file) {
+  while (!fs.existsSync(file)) {
+    await sleep(50);
+  }
+  return require(file);
+}
+/**
+ * Generate a sidebar object for use as reference documentation sidebar entry for each package.
+ * @param {string} pkg the sub-package to generate refdoc options for
+ */
+async function packageSidebarEntry(pkg) {
+  return {
+    type: 'category',
+    label: `@slack/${pkg}`,
+    link: {
+      type: 'doc',
+      id: `reference/${pkg}/index`,
+    },
+    items: await waitAndImportFile(`./content/reference/${pkg}/typedoc-sidebar.cjs`),
```

Comment:
One issue I saw playing around with the docs is that sometimes, as part of docs re-generation, I would get `MODULE_NOT_FOUND` errors where the sidebar generation was executing, and the generated typedoc files for each sub-package was missing. I _think_ this has something to do with the docusaurus lifecycle: plugins we use, like typedoc, are still executing will docusaurus is trying to generate the sidebar. This lifecycle is a bit unclear to me. In any case, to prevent this error, I rewrote this part so that `require`ing the relevant sub-package's generated typedoc only happens once the generated file is actually present. It seems to have addressed the issue.

Reply:
is the file pretty much guaranteed to generate within 50ms or should we increase this for flexibility?